### ISSUE TITLE
flow-typed react

### ISSUE THREAD

**Issue by tomitrescak** [flowtype/flow-typed#259]
Hi, I tried to convert the d.ts of react library to flow-typed but I failed as flow has less constructs. Is there any typing available for react? Thanks!

**Comment by thangngoc89** [comment]
I think React is officially supported by Flow. See https://flowtype.org/docs/react.html

**Comment by ryyppy** [comment]
Exactly, for now the react libdef is in the `flow` project itself.
You can find the definitions here (if you are interested in the internals): https://github.com/facebook/flow/blob/master/lib/react.js

**Comment by tomitrescak** [comment]
Thanks.

**Comment by tomitrescak** [comment]
Hmmm, this does not seem to work. The moment I put node_modules into [ignore] part. React no longer recognises React.Component. Is that normal?

I even tried to copy that definition into my libs folder to no avail.

What I noticed is that they defined the lib as `declare react`. In m experience this does not work and needs to be defined as `declare 'react'`. Anyone managed to solve this?

**Comment by ryyppy** [comment]
> React no longer recognises React.Component

Do you have a repo to show the faulty code? Usually you don't need to do configure anything. And ignoring `node_modules` shouldn't effect you either?

In case it is useful, here is a working example of mine: https://github.com/runtastic/flow-guide/blob/master/tutorial/01-react/00-intro.js

**Comment by tomitrescak** [comment]
@ryyppy Thanks for your kind answer. 
I have nailed the culprit but it looks like a big in flow. I have created a reproducible repo here: https://github.com/tomitrescak/flow-crash-example.git

I know its a bit of files, but the only file one needs to focus on is the "declarations/semanticui-react.js".

When you compile the project now you will see no errors (only eslint crashes on the declaration of the class), but when you try to make eslint happy and you change 

```
import React from 'react';
```

to

```
import { Component } from 'react'; 
```

Flow stops to recognise Component in React and throws several errors. Any idea what could be going wrong?
